File: src/main/java/org/cerion/stocklist/arrays/ValueArray.java
package org.cerion.stocklist.arrays;

public abstract class ValueArray {

	public abstract int size();

	//Use a lower period value when calculating array elements before that position so all values get set to something
	public static int maxPeriod(int pos, int period)
	{
		return (pos < period-1 ? pos+1 : period);
	}

}

File: src/main/java/org/cerion/stocklist/arrays/ValueArray.kt
package org.cerion.stocklist.arrays

abstract class ValueArray {

    abstract val size: Int

    // TODO remove after other classes are refactored
    protected fun size(): Int {
        return size
    }

    companion object {
        //Use a lower period value when calculating array elements before that position so all values get set to something
        fun maxPeriod(pos: Int, period: Int): Int {
            return if (pos < period - 1) pos + 1 else period
        }
    }

}
